figures/scripts/supp-S3-panels-BCD.R

```diff
@@ -0,0 +1,321 @@
+    color_names %in% c("divergent_low_4", "divergent_neutral", "divergent_high_4")
+  ) %>%
+  pull(hex_codes)
+
+
+# Read in metadata and merge with colors
+metadata <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"), guess_max = 10000) %>%
+  # Join on the colors 
+  left_join(histology_label_mapping, by = "cancer_group") %>%
+  # Select what is needed
+  select(contains("cancer_group"), 
+                Kids_First_Biospecimen_ID, 
+                tumor_ploidy) %>%
+  # remove "Other"
+  filter(cancer_group != "Other")
+
+
+### Set up chromosomal sizes for making bins. 
+chr_sizes <- read_tsv(file.path(data_dir, "WGS.hg38.strelka2.unpadded.bed"),
+                             col_names = c("chrom", "start", "end")
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  mutate(chrom = factor(gsub("chr", "", chrom),
+                               levels = c(1:22, "X", "Y", "M")
+  )) %>%
+  # Remove sex chromosomes
+  filter(!(chrom %in% c("X", "Y", "M")))
+
+
+
+
+
+########### Set up heatmap annotation objects
+
+
+# Make chromosome size named vector for Heatmap annotation
+chr_sizes_vector <- chr_sizes$end
+names(chr_sizes_vector) <- chr_sizes$chrom
+
+# Get a vector of the biospecimen IDs we use in the heatmap
+sample_ids <- bin_calls_df$biospecimen_id
+
+
+# Make color key. 
+color_key <- structure(
+  c(divergent_col_hex, "#9932CC", "#fed8b1"),
+  names = c("loss", "neutral", "gain", "unstable", "uncallable")
+)
+
+
+### Make chromosome labeling `HeatmapAnnotation` object
+
+# First, set up bins of ~1Mb size and uncompress
+bins <- GenomicRanges::tileGenome(
+  chr_sizes_vector,
+  tilewidth = 1e6
+)
+bins <- unlist(bins)
+# use to set up chrs
+chrs <- paste0("chr", S4Vectors::decode(bins@seqnames))
+chrs <- factor(chrs, levels = paste0("chr", 1:22))
+
+# Make a key for assigning alternating colors to the chromosomes
+chr_colors <- rep(c("white", "black"), length.out = length(unique(chrs)))
+names(chr_colors) <- unique(chrs)
+
+# Get coordinate start, end, midpoint positions
+chr_start <- match(unique(chrs), chrs)
+chr_end <- chr_start + summary(chrs)
+mid_points <- floor((chr_start + chr_end) / 2)
+
+# Make text labels for chromosome text
+chr_text <- ComplexHeatmap::anno_mark(
+  at = mid_points,
+  labels = levels(chrs),
+  which = "column",
+  side = "bottom",
+  labels_rot = 45,
+  labels_gp = grid::gpar(cex = 0.65)
+)
+
+# Create the Heatmap annotation object
+chr_annot <- HeatmapAnnotation(
+  df = data.frame(chrs),
+  col = list(chrs = chr_colors),
+  name = "",
+  show_legend = FALSE,
+  show_annotation_name = FALSE,
+  mark = chr_text, # Put the text in
+  border = TRUE
+)
+
+
+
+### Make row annotation object
+# Get the histologies for the samples in this set and order them by histology
+
+# Figure out cancer group counts, where NA display groups are _excluded_
+cancer_group_counts <- metadata %>%
+  drop_na(cancer_group_display) %>%
+  # No other ? TODO!
+  #filter(cancer_group_display != "Other") %>%
```

Comment:
Can probably take this out now, since you've filtered `metadata` to remove `"Other"` already.